components/http_foundation.rst

```diff
@@ -712,6 +712,34 @@ Session
 
 The session information is in its own document: :doc:`/components/http_foundation/sessions`.
 
+UrlHelper
+-------
+
+Generating absolute (and relative) URLs for a given path is a common need in lots of applications. In Twig templates this is trivial thanks to the absolute_url() and relative_path() functions. The same functionality can be found in the :class:`Symfony\\Component\\HttpFoundation\\UrlHelper` class, which can be injected as a service anywhere in your application. This class provides two public methods called getAbsoluteUrl() and getRelativePath()::
+
+    // src/Normalizer/UserApiNormalizer.php
+
+    namespace App\Normalizer;
+
+    use Symfony\Component\HttpFoundation\UrlHelper;
+
+    class UserApiNormalizer
+    {
+        private UrlHelper $urlHelper;
+
+        public function __construct(UrlHelper $urlHelper)
+        {
+            $this->urlHelper = $urlHelper;
+        }
+
+        public function normalize($user, $format = null, array $context = [])
```

Comment:
```suggestion
        public function normalize($user): array
```

as seems not used :)
do not know if we can typehint the $user tho

Reply:
d829020